src/seedu/addressbook/data/person/Person.java

```diff
@@ -34,6 +36,19 @@ public Person(ReadOnlyPerson source) {
         this(source.getName(), source.getPhone(), source.getEmail(), source.getAddress(), source.getTags());
     }
 
+    public void setName(String newName) throws IllegalValueException {
```

Comment:
I think for better encapsulation, it would be better to put these methods into the individual classes? For example, setName() in Name class and setEmail() in Email class
Or changing the name in Person to edit...() and the individual classes to set...().
(Or maybe it does not matter and they're equivalent)